lib/view_model/dashboard/home_settings_view_model.dart

```diff
@@ -144,10 +144,12 @@ abstract class HomeSettingsViewModelBase with Store {
 
     if (_balanceViewModel.wallet.type == WalletType.ethereum) {
       ethereum!.addErc20Token(_balanceViewModel.wallet, token as Erc20Token);
+      if (!value) ethereum!.removeTokenTransactionsInHistory(_balanceViewModel.wallet, token);
     }
 
     if (_balanceViewModel.wallet.type == WalletType.polygon) {
       polygon!.addErc20Token(_balanceViewModel.wallet, token as Erc20Token);
+      if (!value) polygon!.removeTokenTransactionsInHistory(_balanceViewModel.wallet, token);
```

Comment:
no need to modify anything, but just to let you know, you could have called the `removeTokenTransactionsInHistory` function from the already existing `addErc20Token` function in `evm_chain_wallet.dart` without having to add it in configure and in cw_evm and making it public